### PR TITLE
CAS-9839 inaccurate gridline plotting, use casacore to parse file header and feed to AST

### PR DESCRIPTION
**Pull request by CNOCycle**
This pull request is related to #21.

To solve various FITS format, FITS header parsing which is implemented by manually is replaced with casacore. Now, CARTA handles all kind of FITS correctly except for rotated image with PC matrix.

In AST library, the definition of coordinate system is slightly different from casacore. `J2000` and `B1950` system should be set `system=FK5,equinox=J2000` and `system=FK4,equinox=B1950`.

CARTA shows correctly gridline in native coordinate system which is defined in FITS header, but CARTA may not get a high-precision gridline if coordinate system is converted. I guess that there are some fundamental limitations on FITS header and coordinate transformation algorithm.

There are a lot of unused code which have been removed in the branch `WCSPlotter/HeaderPaser`. It will be able to merge to branch `develop` if NRAO fix the issues with PC matrix.

### CODE REVIEW

File: carta/cpp/core/Data/Image/LayerData.cpp [hunk 1]
```diff
@@ -663,7 +663,8 @@ void LayerData::_load(std::vector<int> frames, bool recomputeClipsOnNewFrame,
         if ( m_dataGrid ){
             if ( m_dataGrid->_isGridVisible() ){
                 std::shared_ptr<Carta::Lib::IWcsGridRenderService> gridService = m_dataGrid->_getRenderer();
-                gridService->setInputImage( m_dataSource->_getImage() );
+                //gridService->setInputImage( m_dataSource->_getImage() );
+                gridService->setInputImage( m_dataSource->_getPermImage() );
```
Comment (grimmerk) [inline carta/cpp/core/Data/Image/LayerData.cpp:667]: @CNOCycle , need more explanation. 
1. why change to m_dataSource->_getPermImage()

File: carta/cpp/core/Data/Image/Grid/LabelFormats.cpp [hunk 2]
```diff
@@ -69,7 +69,7 @@ QString LabelFormats::getDefaultFormat( const QString& direction ) const {
 
 QString LabelFormats::getDefaultFormatForAxis( Carta::Lib::AxisInfo::KnownType axis,
         const Carta::Lib::KnownSkyCS& cs ) const {
-    QString defaultFormat = FORMAT_DECIMAL_DEG;
+    QString defaultFormat = FORMAT_DECIMAL;
```
Comment (grimmerk) [inline carta/cpp/core/Data/Image/Grid/LabelFormats.cpp:72]: @CNOCycle , need more explanation.

File: carta/cpp/plugins/WcsPlotter/FitsHeaderExtractor.h [hunk 3]
```diff
@@ -19,25 +19,54 @@ class FitsHeaderExtractor
...
 
     /// \return list of errors (if any)
-    QStringList
-    getErrors();
+    QStringList getErrors();
 
 private:
 
-    QStringList tryRawFits( QString fname);
-    QStringList tryCasaCoreFitsConverter( casa::LatticeBase * lbase);
```
Comment (grimmerk) [inline carta/cpp/plugins/WcsPlotter/FitsHeaderExtractor.h:39]: What is the purpose of tryRawFits, tryCasaCoreFitsConverter? no use?

**Review by grimmerk — COMMENTED** [review]
@CNOCycle , need your help to explain some questions and introduce key point of these commits.

**Comment by grimmerk** [comment]
In https://github.com/CARTAvis/carta/pull/21, 
What does this mean? 
CasaImage does not follow FITS-WCS standard, so I'm trying to fix it
CasaImage format -> PC%2d_%2d
but,
FITS-WCS standard -> PC%1d_%1d

@CNOCycle : 2d means 2 digit.

**Comment by kswang1029** [comment]
This means only 1 digit is needed. That is PC0_1, not PC00_01

File: carta/cpp/plugins/WcsPlotter/FitsHeaderExtractor.cpp [hunk 4]
```diff
@@ -127,384 +154,52 @@ FitsHeaderExtractor::tryCasaCoreFitsConverter( casa::LatticeBase * lbase )
...
+                                                       BITPIX, minPix, maxPix,
+                                                       degenerateLast, verbose,
+                                                       stokesLast, preferWavelength,
+                                                       airWavelength,
+                                                       primHead, allowAppend,
+                                                       originStr, history);
+
+    if(!isok)
+    {
+        m_errors << originStr.c_str();
```
Comment (grimmerk) [inline carta/cpp/plugins/WcsPlotter/FitsHeaderExtractor.cpp:178]: @CNOCycle : should correct originStr ->errorString

File: carta/cpp/plugins/WcsPlotter/FitsHeaderExtractor.h (hunk 3 above)
Comment (grimmerk) [reply carta/cpp/plugins/WcsPlotter/FitsHeaderExtractor.h:39]: @CNOCycle 
Old way
Try  tryRawFits first, if fail, try to use tryCasaCoreFitsConverter for treat casa image file/miriad file.

File: carta/cpp/core/Data/Image/Grid/LabelFormats.cpp (hunk 2 above)
Comment (grimmerk) [reply carta/cpp/core/Data/Image/Grid/LabelFormats.cpp:72]: FORMAT_DECIMAL_DEG is about the format of axis. But some axes is not about Degree.

File: carta/cpp/core/Data/Image/LayerData.cpp (hunk 1 above)
Comment (grimmerk) [reply carta/cpp/core/Data/Image/LayerData.cpp:667]: @CNOCycle , since after permutating, ccimage.h did change its header info as permutated status.

**Comment by grimmerk** [comment]
Note: 
Use #42 to merge its and this pull request' content. #40 is for reviewing. Do not merge.

**Pull request merged**